### PR TITLE
Add_item_type_17/18/19

### PR DESCRIPTION
**Pull request by david-moreau**

### CODE REVIEW

File: item.go [hunk 1]
```diff
@@ -54,6 +54,12 @@ const (
 	Calculated ItemType = 15
 	// JMXAgent type
 	JMXAgent ItemType = 16
+	// SNMPTrap type
+	SNMPTrap ItemType = 17
+	// DEPENDENTItem type
+	DEPENDENTItem ItemType = 18
```
Comment (pdecat) [inline item.go]:
```suggestion
	// DependentItem type
	DependentItem ItemType = 18
```

**Review by xp-1000 — CHANGES_REQUESTED** [review]
these types should also be added to `item_protoype.go` and `lld_rule.go` resources

File: item.go [hunk 2]
```diff
@@ -54,6 +54,12 @@ const (
 	Calculated ItemType = 15
 	// JMXAgent type
 	JMXAgent ItemType = 16
+	// SNMPTrap type
```
Comment (pdecat) [inline item.go]:
```suggestion
	// SNMPTrap type (new in 2.2)
```
https://www.zabbix.com/documentation/2.2/manual/api/reference/item/object

File: item.go [hunk 3]
```diff
@@ -54,6 +54,12 @@ const (
...
+	SNMPTrap ItemType = 17
+	// DependentItem type
```
Comment (pdecat) [inline item.go]:
```suggestion
	// DependentItem type (new in 3.4)
```
https://www.zabbix.com/documentation/3.4/manual/api/reference/item/object

File: item.go [hunk 4]
```diff
@@ -54,6 +54,12 @@ const (
...
+	DependentItem ItemType = 18
+	//HTTPAgent type
```
Comment (pdecat) [inline item.go]:
```suggestion
	//HTTPAgent type (new in 4.0)
```
https://www.zabbix.com/documentation/4.0/manual/api/reference/item/object

**Pull request merged**